cloud_governance/common/clouds/azure/cost_management/cost_management_operations.py

```diff
@@ -15,54 +16,64 @@ def __init__(self):
-            end_date = (current + datetime.timedelta(60))
-            month_end = calendar.monthrange(end_date.year, end_date.month)[1]
-            end_date = end_date.replace(day=month_end)
-        logger.info(f'StartDate: {start_date}, EndDate: {end_date}')
-        response = self.azure_operations.cost_mgmt_client.forecast.usage(scope=self.azure_operations.scope, parameters={
-            'type': 'ActualCost', 'timeframe': 'Custom',
-            'time_period': QueryTimePeriod(from_property=start_date, to=end_date),
-            'dataset': QueryDataset(granularity=granularity, aggregation={
-                "totalCost": QueryAggregation(name="Cost", function="Sum"),
-            }), 'include_actual_cost': True, 'include_fresh_partial_cost': False
-        }).as_dict()
-        result = {'columns': response.get('columns'), 'rows': []}
-        row_data = {}
-        for data in response.get('rows'):
-            data_date = data[1]
-            if data_date in row_data:
-                if row_data[data_date][2] == 'Actual' and data[2] == 'Forecast':
-                    row_data[data_date][2] = data[2]
-                row_data[data_date][0] += data[0]
-            else:
-                row_data[data_date] = data
-        result['rows'] = list(row_data.values())
-        return result
+        try:
+            if not start_date and not end_date:
+                current = datetime.datetime.now(pytz.UTC)
+                start_date = (current - datetime.timedelta(60)).replace(day=1)
+                end_date = (current + datetime.timedelta(365))
+                month_end = calendar.monthrange(end_date.year, end_date.month)[1]
+                end_date = end_date.replace(day=month_end)
+            logger.info(f'StartDate: {start_date}, EndDate: {end_date}')
+            response = self.azure_operations.cost_mgmt_client.forecast.usage(scope=scope, parameters={
+                'type': 'ActualCost', 'timeframe': 'Custom',
+                'time_period': QueryTimePeriod(from_property=start_date, to=end_date),
+                'dataset': QueryDataset(granularity=granularity, aggregation={
+                    "totalCost": QueryAggregation(name="Cost", function="Sum"),
+                }, **kwargs), 'include_actual_cost': True, 'include_fresh_partial_cost': False
+            }).as_dict()
+            result = {'columns': response.get('columns'), 'rows': []}
+            row_data = {}
+            for data in response.get('rows'):
+                data_date = data[1]
+                if data_date in row_data:
+                    if row_data[data_date][2] == 'Actual' and data[2] == 'Forecast':
+                        row_data[data_date][2] = data[2]
+                    row_data[data_date][0] += data[0]
+                else:
+                    row_data[data_date] = data
+            result['rows'] = list(row_data.values())
+            return result
+        except Exception as err:
+            time.sleep(60)
```

Comment:
Why we need this sleep after error ?
Can you put 60 as  constant ?

Reply:
When we continuously start fetching, will get the limit exceeded. So we need to wait sometime.
ok will make constant

Reply:
OK, so just make it constant SLEEP_TIME